_docs/feelgood.md

```diff
@@ -8,4 +8,145 @@ permalink: /feelgood.html
+  [API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-getting-started-with-rest-apis.html).
+* The Lambda function takes the JSON payload delivered by the POST request, 
+  which contains the user's phone number and selected topic. The function 
+  then:
+    * Validates the phone number for correct formatting 
+    * Looks up the [Amazon Resource Name](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) 
+      (ARN) for the selected topic
+    * Invokes the [Simple Notification Service](https://aws.amazon.com/sns/) SDK
+    * Makes a `subscribe()` call to SNS, passing the user's data and topic ARN
+* SNS saves the user's phone number as a subscriber under the specified topic.
+* If the Lambda function processes successfully, it will return a success code 
+  to the API Gateway method, which in turn returns the HTTP 200 status to the 
+  user's browser. The Lambda function also publishes a text message to the user's 
+  provided number, which confirms subscription and welcomes the user.
+
+[View the website code here.](https://github.com/UWB-ACM/feelgood/blob/master/website/)
+
+[View the subscription Lambda function here.](https://github.com/UWB-ACM/feelgood/blob/master/subscribe/lambda_function.py)
+
+Here's a diagram of the system's flow of logic:
+
+![FeelGood subscription mechanism](/assets/feelgood_sub.png)
+
+### User Notification
+
+* To send messages out at regular intervals, we used CloudWatch Events 
+  to create [rules which get invoked on a time schedule](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/Create-CloudWatch-Events-Scheduled-Rule.html).
+* When creating the Event rule, the message publication function 
+  [was selected as the target](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/RunLambdaSchedule.html).
+* Upon being invoked, the Lambda function:
+    * Chooses 4 messages to publish (one for each possible topic)
+    * Looks up the ARN for each topic
+    * Invokes the [Simple Notification Service](https://aws.amazon.com/sns/) SDK
+    * Makes 4 `publish()` calls to SNS, where each call uses a different message and topic ARN
+* SNS receives the SDK invocation to publish a message and distributes 
+  the message for each topic to all subscribers for that topic. Each topic 
+  is handled asynchronously.
+
+[View the publication Lambda function here.](https://github.com/UWB-ACM/feelgood/blob/master/subscribe/lambda_function.py)
+
+Here's a diagram of the subsystem's flow of logic:
+
+![FeelGood publication mechanism](/assets/feelgood_pub.png)
+  
+## Implementation Notes
+
+The specific instructions, how-tos, and details we encountered are organized 
+here by AWS product.
+
+### S3 & Website Construction
+
+More info coming soon.
+
+### API Gateway
+
+More info coming soon.
+
+### Lambda
+
+More info coming soon.
+
+### SNS
+
+More info coming soon.
+
+### CloudWatch
+
+More info coming soon.
+
+## Gotchas & Lessons Learned
+
```

Comment:
@anishprasad01 @tcumings adding your implementation notes in here first will make duplicating info in the code repo (https://github.com/UWB-ACM/feelgood) easier, since we can copy and paste this entire file and associated images.